### PR TITLE
[MBL-17285][Student][Teacher] -  Implement Calendar Event Screen E2E test

### PR DESCRIPTION
**Pull request by kdeakinstructure**
Implement Student Calendar Event create/edit/delete test.
Implement Teacher Calendar Event create/edit/delete test.

Add new page objects (screen pages, event details pages).
Add lots of testTags on composables in order to make them more easy to select with matchers.
Import compose UI dependency in espresso build.gradle
Minor refactor, renaming on existing methods.
Add some utility functions.
Add new FeatureCategory.

Restructure compose UI pages and e2e tests into 'compose' packages.

refs: MBL-17285
affects: Student, Teacher
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt [hunk 1]
```diff
@@ -231,21 +233,21 @@ private fun EventContent(
...
-                    modifier = Modifier.padding(horizontal = 16.dp),
+                    modifier = Modifier.padding(horizontal = 16.dp).testTag("eventDate"),
                     color = colorResource(id = R.color.textDarkest),
                     fontSize = 16.sp
                 )
                 if (uiState.recurrence.isNotEmpty()) {
                     Text(
                         text = uiState.recurrence,
-                        modifier = Modifier.padding(horizontal = 16.dp),
+                        modifier = Modifier.padding(horizontal = 6.dp).testTag("recurrence"),
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt:250]: Why did you change the padding here?

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt (hunk 1 above)
Comment (kdeakinstructure) [reply libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt:250]: It was just experiment, I forgot to set it back, nice catch!

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: automation/espresso/build.gradle [hunk 2]
```diff
@@ -78,6 +78,8 @@ android {
 }
 
 dependencies {
+    implementation 'androidx.compose.ui:ui-test-android:1.6.6'
```
Comment (kristofnemere) [inline automation/espresso/build.gradle:81]: Would be better to use the constanst to make sure we use the same version at all places: implementation Libs.COMPOSE_UI_TEST

**Review by kristofnemere — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: automation/espresso/build.gradle (hunk 2 above)
Comment (kdeakinstructure) [reply automation/espresso/build.gradle]: Did it, thanks!

**Review by kristofnemere — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Pull request merged**